src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewList.cs

```diff
@@ -0,0 +1,687 @@
+						if (draggedOverItemPeer != null)
+						{
+							draggedOverString = draggedOverItemPeer.GetName(); //TODO: draggedOverItemPeer.GetNameCore();
+						}
+					}
+					else
+					{
+						int insertIndex = -1;
+						int afterInsertIndex = -1;
+						int beforeInsertIndex = -1;
+						var dragItemIndex = IndexFromContainer(dragItem);
+
+						if (keyboardReorderedContainer != null)
+						{
+							insertIndex = dragItemIndex;
+						}
+						else
+						{
+							insertIndex = m_emptySlotIndex;
+						}
+
+						if (insertIndex != -1)
+						{
+							afterInsertIndex = insertIndex + 1;
+							beforeInsertIndex = insertIndex - 1;
+						}
+
+						if (insertIndex > dragItemIndex)
+						{
+							beforeInsertIndex++;
+						}
+						else if (insertIndex < dragItemIndex)
+						{
+							afterInsertIndex--;
+						}
+
+						itemBeforeInsertPositionString = GetAutomationName(beforeInsertIndex);
+						itemAfterInsertPositionString = GetAutomationName(afterInsertIndex);
+					}
+
+					m_dropTargetDropEffectString = BuildEffectString(itemBeforeInsertPositionString, itemAfterInsertPositionString, dragItemString, draggedOverString);
+				}
+
+				if (!forceUpdate && oldValue != m_dropTargetDropEffectString)
+				{
+					AutomationPeer treePeer = FrameworkElementAutomationPeer.FromElement(this);
+					treePeer.RaisePropertyChangedEvent(DropTargetPatternIdentifiers.DropTargetEffectProperty, oldValue, m_dropTargetDropEffectString);
+				}
+			}
+		}
+
+		internal void EnableMultiselect(bool isEnabled)
+		{
+			m_isMultiselectEnabled = isEnabled;
+		}
+
+		internal bool IsMultiselect => m_isMultiselectEnabled;
+
+		internal bool IsMutiSelectWithSelectedItems
+		{
+			get
+			{
+				var selectedItems = ListViewModel.SelectedNodes;
+				bool isMutiSelect = m_isMultiselectEnabled && selectedItems.Count > 0;
+				return isMutiSelect;
+			}
+		}
+
+		private bool IsSelected(TreeViewNode node)
+		{
+			bool isSelected = false;
+			var selectedItems = ListViewModel.SelectedNodes;
+			for (var i = 0; i < selectedItems.Count; i++)
+			{
+				if (selectedItems[i] == node)
+				{
+					isSelected = true;
+					break;
+				}
+			}
+
+			return isSelected;
+		}
+
+		internal IList<TreeViewNode> GetRootsOfSelectedSubtrees()
+		{
+			var roots = new List<TreeViewNode>();
+			var selectedItems = ListViewModel.SelectedNodes;
+			for (var i = 0; i < selectedItems.Count; i++)
+			{
+				var item = selectedItems[i];
+				var selectionRoot = GetRootOfSelection(item);
+
+				if (!roots.Contains(selectionRoot))
+				{
+					roots.Add(selectionRoot);
+				}
+			}
+
+			return roots;
+		}
+
+		internal int FlatIndex(TreeViewNode node)
+		{
+			bool found = ListViewModel.IndexOfNode(node, out var flatIndex);
+			flatIndex = found ? flatIndex : -1;
+			return flatIndex;
+		}
+
+		internal bool IsFlatIndexValid(int index)
+		{
+			return index >= 0 && index < ListViewModel.Size;
+		}
+
+		private int RemoveNodeFromParent(TreeViewNode node)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change return type to 'void'; not a single caller uses the returned value.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)